crates/starknet/types/src/lib.rs

```diff
@@ -88,13 +88,17 @@ impl From<Call> for starknet::core::types::Call {
 
 /// Validates that a Felt value represents a valid Starknet contract address.
 ///
-/// ### Arguments
-/// * `felt` - The Felt value to validate as a Starknet address
+/// In Starknet, contract addresses must follow specific constraints to be considered valid:
+/// - They must be greater than or equal to 2, as addresses 0 and 1 are reserved for system use:
+///   * 0x0 acts as the default caller address for external calls and has no storage
+///   * 0x1 functions as a storage space for block mapping [link](https://docs.starknet.io/architecture-and-concepts/network-architecture/starknet-state/#special_addresses)
+/// - They must be less than 2^251 (0x800000000000000000000000000000000000000000000000000000000000000)
+///   as this is the upper bound of the Starknet field prime
```

Comment:
It is not the bound of the prime, which. is 2^251 + 17 * 2^192.
Here it is 2^251, which is sometimes called `ELEMENT_UPPER_BOUND`. But it is never explained precisely in the doc why it is this value.

I think it is due to the way the storage merkle tree is constructed. But not 100% sure